--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -31,7 +31,8 @@
 #include <linux/sec_debug.h>
 
 DEFINE_PER_CPU_SHARED_ALIGNED(struct rq, runqueues);
-<<<<<<< HEAD
+DEFINE_PER_CPU(struct rnd_state, sched_rnd_state);
+DEFINE_PER_CPU_READ_MOSTLY(u64, dvfs_update_delay);
 
 #ifdef CONFIG_SCHED_DEBUG
 /*
@@ -48,10 +49,6 @@
 	0;
 #undef SCHED_FEAT
 #endif
-=======
-DEFINE_PER_CPU(struct rnd_state, sched_rnd_state);
-DEFINE_PER_CPU_READ_MOSTLY(u64, dvfs_update_delay);
->>>>>>> 9ccaeb959240 (BACKPORT: sched: cpufreq: Remove magic 1.25 headroom from apply_dvfs_headroom())
 
 /*
  * Number of tasks to iterate in a single balance run.
@@ -1081,6 +1078,12 @@
 		uclamp_rq_dec_id(rq, p, clamp_id);
 }
 
+static inline unsigned long uclamp_rq_get(struct rq *rq,
+					   enum uclamp_id clamp_id)
+{
+	return rq->uclamp[clamp_id].value;
+}
+
 static inline void
 uclamp_update_active(struct task_struct *p)
 {
@@ -1372,6 +1375,11 @@
 #else /* CONFIG_UCLAMP_TASK */
 static inline void uclamp_rq_inc(struct rq *rq, struct task_struct *p) { }
 static inline void uclamp_rq_dec(struct rq *rq, struct task_struct *p) { }
+static inline unsigned long uclamp_rq_get(struct rq *rq,
+					   enum uclamp_id clamp_id)
+{
+	return clamp_id == UCLAMP_MIN ? 0 : SCHED_CAPACITY_SCALE;
+}
 static inline int uclamp_validate(struct task_struct *p,
 				  const struct sched_attr *attr)
 {
