--- a/kernel/sched/walt.c
+++ b/kernel/sched/walt.c
@@ -2021,7 +2021,7 @@
 
 	if (!use_cycle_counter) {
 		rq->task_exec_scale = DIV64_U64_ROUNDUP(cpu_cur_freq(cpu) *
-				topology_get_cpu_scale(NULL, cpu),
+				topology_get_cpu_scale(cpu),
 				rq->cluster->max_possible_freq);
 		return;
 	}
@@ -2058,7 +2058,7 @@
 		SCHED_BUG_ON((s64)time_delta < 0);
 
 		rq->task_exec_scale = DIV64_U64_ROUNDUP(cycles_delta *
-				topology_get_cpu_scale(NULL, cpu),
+				topology_get_cpu_scale(cpu),
 				time_delta * rq->cluster->max_possible_freq);
 		trace_sched_get_task_cpu_cycles(cpu, event,
 				cycles_delta, time_delta, p);
@@ -2232,7 +2232,7 @@
 
 #define pct_to_min_scaled(tunable) \
 		div64_u64(((u64)sched_ravg_window * tunable *		\
-			 topology_get_cpu_scale(NULL,			\
+			 topology_get_cpu_scale(			\
 			 cluster_first_cpu(sched_cluster[0]))),	\
 			 ((u64)SCHED_CAPACITY_SCALE * 100))
 
@@ -2333,7 +2333,7 @@
 
 	raw_spin_lock_init(&cluster->load_lock);
 	cluster->cpus = *cpus;
-	cluster->efficiency = topology_get_cpu_scale(NULL, cpumask_first(cpus));
+	cluster->efficiency = topology_get_cpu_scale(cpumask_first(cpus));
 
 	if (cluster->efficiency > max_possible_efficiency)
 		max_possible_efficiency = cluster->efficiency;
